src/ceph-volume/ceph_volume/util/encryption.py

```diff
@@ -60,6 +61,63 @@ def get_sector_size(device):
 
     return sector_size
 
+def is_rotational(device):
+    """
+    Test if device is rotational or not
+
+    :param device: Absolute path to device
+    """
+    device_metadata = lsblk(device, abspath=True)
+    rota_state = device_metadata['ROTA']
+    rota_state = True if rota_state == '1' else False
+
+    return rota_state
+
+def execCmd(cmdline):
+    cmd = subprocess.Popen(cmdline, shell=True, stdout=subprocess.PIPE)
+    stdout, _ = cmd.communicate()
+    return stdout.decode("utf-8")
+
+def bypass_workqueues(device):
+    """
+    1) Check if cryptsetup has support for no_read_workqueue.
+    If so, it also supports no_write_workqueue.
+    cryptsetup has support for this since 2.3.4
+
+    2) Also check if we are using a flash (non-rotational) device
+    or not. I.e. Spinning media might still benefit from queuing.
+
+    Only if both 1) and 2) are true we do _not_ want to use work queues
+    Note:
+    crypt version 1.22 and higher have support for this. Example:
+    modprobe dm-crypt
+    dmsetup targets
+
+    integrity        v1.10.0
+    crypt            v1.23.0
+    striped          v1.6.0
+    linear           v1.4.0
+    error            v1.5.0
+
+    If dm-crypt does not have support for no_read_workqueue /
+    no_write_workqueue the options will be discarded and
+    logged by the linux kernel:
+
+    device-mapper: table: major:minor: crypt: Invalid feature arguments
+    device-mapper: ioctl: error adding target to table
+
+    The encrypted device will come online without the options active.
+    """
+    noworkqueue_support = False
+    cryptsetup_help = execCmd("cryptsetup --help 2>/dev/null")
+    if "--perf-no_read_workqueue" in cryptsetup_help:
+        noworkqueue_support=True
+
+        if is_rotational(device) is False:
```

Comment:
To be more Pythonic, this should be `if not is_rotational(device):`. But given the above comment about the unused var, this could probably all be simplified to something like:
```python
     if "--perf-no_read_workqueue" not in cryptsetup_help or is_rotational(device):
         return []

     return ['--perf-no_read_workqueue', '--perf-no_write_workqueue']
```

Reply:
Yes, this has been rewritten as you suggested.

Reply:
Thanks!